README.md

```diff
@@ -1 +1,191 @@
+
+In most applications this may not be a problem, but if we need a good user experience on devices with fewer resources, for example on mobile device, we may want to avoid unnecessary calculations.
+
+> NOTE: we can simulate it using the `Performance` tab in the browser dev tools.
+
+In that case, we need to use `React.useMemo` to memoize the result of the `getFiltered` function that depends on the `products` and `visibility`:
+
+_./src/app.tsx_
+
+```diff
+...
+
+const ProductList = ({ visibility }) => {
+  const [products, setProducts] = React.useState(INITIAL_PRODUCTS);
+  const handleChange = React.useCallback((id) => {
+    setProducts((products) => getUpdated(id, products));
+  }, []);
+- const filtered = getFiltered(products, visibility);
++ const filtered = React.useMemo(
++   () => getFiltered(products, visibility),
++   [products, visibility]
++ );
+
+...
+
+```
+
+Depending on the device it may go faster or slower, but we have improved somewhat.
+
+## SolidJS
+
+In this example we have implemented the same as the previous one but using SolidJS.
+
+We still use `JSX` but instead of having a React state with the products we use signals. In this case, we don't need to wrap the `handleChange` function with `useCallback` nor the `filtered` with `useMemo` because these functions will be created only once (even if we update the properties of that product).
+
+_./src/app.tsx_
+
+```jsx
+const ProductList = (props) => {
+  const [products, setProducts] = createSignal(INITIAL_PRODUCTS);
+  const handleChange = (id) => {
+    setProducts((products) => getUpdated(id, products));
+  };
+  const filtered = () => getFiltered(products(), props.visibility);
+...
+
+```
+
+Since `JSX` wraps the `createEffect` function, the filtered products will be updated when the `products` signal or visibility prop changes but not in color updates.
+
+_./src/app.tsx_
+
+```jsx
+const ProductList = (props) => {
+  const [products, setProducts] = createSignal(INITIAL_PRODUCTS);
+  const handleChange = (id) => {
+    setProducts((products) => getUpdated(id, products));
+  };
+  const filtered = () => getFiltered(products(), props.visibility);
+
+  return (
+    <>
+      <ul>
+        <For each={filtered()}>
+          {(product) => <Product product={product} onChange={handleChange} />}
+        </For>
+      </ul>
+      <AddProduct onAdd={setProducts} />
+    </>
+  );
+};
+```
+
+> NOTE: Also, we can simulate it using the `Performance` tab in the browser dev tools and we see that the performance is much better.
+
+## Svelte
+
+In this case, components are written into `.svelte` files using a superset of HTML. In the `script` tag, we can write JavaScript/TypeScript code and define the `products` in a JavaScript variable using `let`.
+
+We will change the array reference when we add a new product or change the product status and using the `$` trick we will tell the compiler that the `filtered` variable will be recalculated when the `products` or `visibility` change.
```

Comment:
The product array list will change whenever we add a new product, or whenever the user clicks on the fav/unfav icons (filtering), to control this second case we will ask _Svelte_ to watch this value and execute a given code when that happens (to do that we will use the _$_ sign _Svelte_ this will allow us to define a reactive statement).